helix-core/src/selection.rs

```diff
@@ -535,6 +540,12 @@ impl Selection {
     pub fn len(&self) -> usize {
         self.ranges.len()
     }
+
+    pub fn contains(&self, other: &Selection) -> bool {
+        self.iter()
+            .zip(other.iter())
```

Comment:
Yeah, you are right. I reworked the method but I am still hesitant whether the check makes sense at all. I think just moving the cursor should disable the `shrink_selection` so that we are safe, but I guess we can adjust that later if there are any complaints?

Reply:
I was pondering this. I think just checking containment will do the right thing when the user flow is expand, expand, shrink, shrink, etc, but you could also have the following sequence

```rust
fn foo() {
    let a = 1;
    [a]
}
```
expand

```rust
fn foo() [{
    let a = 1;
    a
}]
```

move down

```rust
fn foo() {
[]  let a = 1;
    a
]
```

select entire file

```rust
[
...
fn foo() {
    let a = 1;
    a
}
...
]
```

shrink selection

```rust
fn foo() [{
    let a = 1;
    a
}]
```